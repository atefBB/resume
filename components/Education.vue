<template>
  <div v-if="educationsData.educations" class="container">
    <h3 id="education-title">
      {{
        educationsData.educations_title
          ? educationsData.educations_title
          : "Education"
      }}
    </h3>
    <el-timeline>
      <el-timeline-item
        v-for="(education, index) in educationsData.educations"
        :key="index"
        :timestamp="education.dates"
        placement="top"
      >
        <el-card>
          <h4>
            {{ education.name }}
            <el-divider direction="vertical" />
            <span>
              <i class="fas fa-map-marker-alt" />
              {{ education.place }}
            </span>
          </h4>
          <p>{{ education.qualification }}</p>
        </el-card>
      </el-timeline-item>
    </el-timeline>
  </div>
</template>

<script>
import educationsData from "~/assets/yaml/educations.yml"
import { safeLoad } from "js-yaml"

export default {
  props: [],
  data() {
    return {
      educationsData: {}
    }
  },
  created: function() {
    this.getContent()
  },
  methods: {
    getContent() {
      this.educationsData = safeLoad(educationsData)
    }
  }
}
</script>
