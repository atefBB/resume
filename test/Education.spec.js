import faker from "faker"
import { testHelper, mountHelper } from "./helpers"
import Education from "~/components/Education.vue"
jest.mock(`~/assets/yaml/educations.yml`)

describe("Education", () => {
  testHelper(Education)

  describe("Matches randoms values", () => {
    const values = {
      educationsData: {
        educations_title: faker.company.bsAdjective(),
        educations: Array.from(
          { length: faker.random.number({ min: 1, max: 3 }) },
          () => {
            return {
              dates:
                faker.random.number({ min: 1990, max: 2020 }).toString() +
                "-" +
                faker.random.number({ min: 1990, max: 2020 }).toString(),
              name: faker.company.companySuffix(),
              place: faker.address.city(),
              qualification: faker.name.jobTitle()
            }
          }
        )
      }
    }
    const mockFn = jest.fn()
    const options = {
      methods: {
        getContent: mockFn
      }
    }
    const factory = mountHelper(Education)

    it("Should call the mocked function", () => {
      factory({ options, values })
      expect(mockFn).toHaveBeenCalled() //we use correctly the mock function
    })

    //educations.length
    it("Should have the same length of timeline and randoms values", () => {
      const wrapper = factory({ options, values })
      const timelineArray = wrapper.findAll(".el-timeline-item__wrapper")
      expect(timelineArray.length).toStrictEqual(
        values.educationsData.educations.length
      )
    })
    it("Should NOT contain the education section", () => {
      let localValues = {
        educationsData: { ...values.about, educations: undefined }
      }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.isEmpty()).toBe(true)
    })

    //educations_title
    it("Should contain the educations_title", () => {
      const wrapper = factory({ options, values })
      expect(wrapper.find("#education-title").text()).toEqual(
        expect.stringContaining(values.educationsData.educations_title)
      )
    })
    it("Should contain the default educations_title", () => {
      let localValues = {
        educationsData: {
          ...values.educationsData,
          educations_title: undefined
        }
      }
      const wrapper = factory({ options, values: localValues })
      expect(wrapper.find("#education-title").text()).toEqual(
        expect.stringContaining("Education")
      )
    })

    //educations.length
    it("Should contain the random values at correct position", () => {
      const wrapper = factory({ options, values })
      const timelineArray = wrapper.findAll(".el-timeline-item__wrapper")
      // check if the values are correctly printed
      for (let index = 0; index < timelineArray.length; index++) {
        const education = values.educationsData.educations[index]
        const wrapperText = timelineArray
          .at(index)
          .find(".el-timeline-item__wrapper")
          .text()
        expect(wrapperText).toEqual(expect.stringContaining(education.dates))
        expect(wrapperText).toEqual(expect.stringContaining(education.name))
        expect(wrapperText).toEqual(expect.stringContaining(education.place))
        expect(wrapperText).toEqual(
          expect.stringContaining(education.qualification)
        )
      }
    })
  })
})
