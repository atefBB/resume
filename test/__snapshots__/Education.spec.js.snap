// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education Default Template 1`] = `
<div class="container">
  <h3 id="education-title">
    Education
  </h3>
  <ul class="el-timeline">
    <li class="el-timeline-item">
      <div class="el-timeline-item__tail"></div>
      <div class="el-timeline-item__node el-timeline-item__node--normal el-timeline-item__node--">
        <!---->
      </div>
      <!---->
      <div class="el-timeline-item__wrapper">
        <div class="el-timeline-item__timestamp is-top">
          2010 - 2012
        </div>
        <div class="el-timeline-item__content">
          <div class="el-card is-always-shadow">
            <!---->
            <div class="el-card__body">
              <h4>
                EPSI
                <div class="el-divider el-divider--vertical">
                  <!---->
                </div> <span><i class="fas fa-map-marker-alt"></i>
            Lyon
          </span></h4>
              <p>Master’s degree</p>
            </div>
          </div>
        </div>
        <!---->
      </div>
    </li>
    <li class="el-timeline-item">
      <div class="el-timeline-item__tail"></div>
      <div class="el-timeline-item__node el-timeline-item__node--normal el-timeline-item__node--">
        <!---->
      </div>
      <!---->
      <div class="el-timeline-item__wrapper">
        <div class="el-timeline-item__timestamp is-top">
          2009 - 2010
        </div>
        <div class="el-timeline-item__content">
          <div class="el-card is-always-shadow">
            <!---->
            <div class="el-card__body">
              <h4>
                Partner Formation
                <div class="el-divider el-divider--vertical">
                  <!---->
                </div> <span><i class="fas fa-map-marker-alt"></i>
            Lyon
          </span></h4>
              <p>Bachelor’s degree « IT Project Manager »</p>
            </div>
          </div>
        </div>
        <!---->
      </div>
    </li>
    <li class="el-timeline-item">
      <div class="el-timeline-item__tail"></div>
      <div class="el-timeline-item__node el-timeline-item__node--normal el-timeline-item__node--">
        <!---->
      </div>
      <!---->
      <div class="el-timeline-item__wrapper">
        <div class="el-timeline-item__timestamp is-top">
          2007 - 2009
        </div>
        <div class="el-timeline-item__content">
          <div class="el-card is-always-shadow">
            <!---->
            <div class="el-card__body">
              <h4>
                Maestris
                <div class="el-divider el-divider--vertical">
                  <!---->
                </div> <span><i class="fas fa-map-marker-alt"></i>
            Valence
          </span></h4>
              <p>HND Application developer</p>
            </div>
          </div>
        </div>
        <!---->
      </div>
    </li>
  </ul>
</div>
`;
